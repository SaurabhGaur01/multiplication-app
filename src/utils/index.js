const MAX_NUMBER = 144;

// will return all divisible numbers of the given number whenever called
export const getAllDivisibleNumbers = number => (
    getDynamicArrayOfGivenSize().filter(item => item % number === 0)
)    

// will return the dynamic array of the given size whenever called
export const getDynamicArrayOfGivenSize = () =>  [...Array(MAX_NUMBER).keys()];
