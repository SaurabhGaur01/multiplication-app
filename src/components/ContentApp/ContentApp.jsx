import React from 'react';
import { getAllDivisibleNumbers, getDynamicArrayOfGivenSize } from '../../utils'

const ContentApp = () => {
    const [activeIndexes, setActiveIndexes] = React.useState([]);

    const onClickHandler = (index) => {
        const getAllIndexes = getAllDivisibleNumbers(index+1); 
        setActiveIndexes(getAllIndexes);
    }

    return (
        <section>
            <ul className="list">
                {
                    getDynamicArrayOfGivenSize().map(item =>{
                        const className = activeIndexes.includes(item+1) ? 'border border-success list-item active' : 'border list-item'; 
                        return (
                            <li 
                                id={`list-item-${item}`}
                                key={`list-item-${item}`}
                                onClick={() => onClickHandler(item)}
                                className={className}
                            >
                                {item + 1}
                            </li>
                        )
                    })
                }
            </ul>
        </section>
    )
}

export default ContentApp