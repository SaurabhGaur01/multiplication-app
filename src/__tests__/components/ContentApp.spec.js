import React from 'react';
import { shallow } from 'enzyme';
import ContentApp from '../../components/ContentApp/ContentApp';
import * as utils from '../../utils/index';

describe('ContentApp', () => {
    it('should render correctly', () => {
        jest.spyOn(utils, 'getDynamicArrayOfGivenSize').mockImplementation(() => [ 1,2,3,4,5 ]);
        const renderedModule = shallow(<ContentApp />);
        expect(renderedModule).toMatchSnapshot();
    });

    it('should call onCityChange when input changes', () => {
        jest.spyOn(utils, 'getDynamicArrayOfGivenSize').mockImplementation(() => [ 1,2,3 ]);
        const divisibleSpy = jest.spyOn(utils, 'getAllDivisibleNumbers').mockImplementation(() => [ 1 ]);
        const renderedModule = shallow(<ContentApp />);
        renderedModule.find('#list-item-1').at(0).simulate('click');
        expect(divisibleSpy).toHaveBeenCalled();
        expect(divisibleSpy).toHaveBeenCalledWith(2);
    });

    it("should update state on click", () => {
        jest.spyOn(utils, 'getDynamicArrayOfGivenSize').mockImplementation(() => [ 1,2,3 ]);
        const setActiveIndexes = jest.fn();
        const renderedModule = shallow(<ContentApp />);
        const useStateMock = jest.spyOn(React, "useState");
       
        useStateMock.mockImplementation(activeIndexes => [activeIndexes, setActiveIndexes]);
        renderedModule.find('#list-item-1').at(0).simulate('click');
        expect(setActiveIndexes).toBeTruthy();
      });
})