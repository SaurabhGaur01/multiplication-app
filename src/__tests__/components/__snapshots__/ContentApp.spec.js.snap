// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContentApp should render correctly 1`] = `
<section>
  <ul
    className="list"
  >
    <li
      className="border list-item"
      id="list-item-1"
      key="list-item-1"
      onClick={[Function]}
    >
      2
    </li>
    <li
      className="border list-item"
      id="list-item-2"
      key="list-item-2"
      onClick={[Function]}
    >
      3
    </li>
    <li
      className="border list-item"
      id="list-item-3"
      key="list-item-3"
      onClick={[Function]}
    >
      4
    </li>
    <li
      className="border list-item"
      id="list-item-4"
      key="list-item-4"
      onClick={[Function]}
    >
      5
    </li>
    <li
      className="border list-item"
      id="list-item-5"
      key="list-item-5"
      onClick={[Function]}
    >
      6
    </li>
  </ul>
</section>
`;
